src/integrityCheck.js

```diff
@@ -0,0 +1,46 @@
+import fs from 'fs'
+import { saveToAddonList } from './config'
+import { mainWindow } from '../main'
+
+// Verifies that all addon subdirs are present. Updates addon statuses accordingly
+export function integrityCheck (installedAddonsDict, configObj) {
+    let changed = false
+    console.log('Checking install integrity')
+    fetchAllAddonDirs(configObj.addonDir)
+      .then(dirList => {
+        for (const [name] of Object.entries(installedAddonsDict)) { // Checks if addon is installed by making sure all subdirs are found in dirlist     
```

Comment:
`for of` is meant for iterables (arrays), `for in` is for enumerables (objects). But your solution works, its just extra steps

Reply:
integrityCheck.js to be refactored